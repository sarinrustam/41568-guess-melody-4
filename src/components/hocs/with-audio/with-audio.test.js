import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withAudio from "./with-audio.js";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

describe(`withAudio tests`, () => {
  it(`withAudio render correctly`, () => {
    const handlerButtonClick = jest.fn();

    const tree = renderer.create((
      <MockComponentWrapped
        isPlaying={false}
        onPlayButtonClick={handlerButtonClick}
        src={`12`}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
