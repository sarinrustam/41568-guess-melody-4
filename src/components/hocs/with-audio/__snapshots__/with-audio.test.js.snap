// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudio tests withAudio render correctly 1`] = `
<div>
  <audio />
</div>
`;
